src/librustdoc/passes/doc_test_lints.rs

```diff
@@ -74,6 +77,8 @@ pub fn should_have_doc_example(item_kind: &clean::ItemEnum) -> bool {
         | clean::PrimitiveItem(_)
         | clean::KeywordItem(_)
     )
+    && !cx.tcx.hir().attrs(node).lists(sym::doc).has_word(sym::hidden)
+    && !inherits_doc_hidden(cx, node)
```

Comment:
Hmm, would it be easier to add the first check to `inherits_doc_hidden` so you don't have to check it at each call site?

Reply:
I'm not sure because `inherits_doc_hidden` is used only in two places with this change, and the other place needs this information a bit ahead. That'd be a duplicate in such case.